clients/admin-ui/src/features/openid-authentication/OpenIDProviderForm.tsx

```diff
@@ -0,0 +1,217 @@
+import { SerializedError } from "@reduxjs/toolkit";
+import { FetchBaseQueryError } from "@reduxjs/toolkit/dist/query";
+import {
+  Box,
+  Button,
+  ConfirmationModal,
+  Stack,
+  Text,
+  useDisclosure,
+  useToast,
+} from "fidesui";
+import { Form, Formik, FormikHelpers } from "formik";
+import { useMemo } from "react";
+import * as Yup from "yup";
+
+import { CustomSelect, CustomTextInput } from "~/features/common/form/inputs";
+import { getErrorMessage, isErrorResult } from "~/features/common/helpers";
+import { errorToastParams, successToastParams } from "~/features/common/toast";
+import {
+  useCreateOpenIDProviderMutation,
+  useDeleteOpenIDProviderMutation,
+  useUpdateOpenIDProviderMutation,
+} from "~/features/openid-authentication/openprovider.slice";
+import { OpenIDProvider } from "~/types/api";
+
+interface OpenIDProviderFormProps {
+  openIDProvider?: OpenIDProvider;
+  onSuccess?: (openIDProvider: OpenIDProvider) => void;
+}
+
+export interface OpenIDProviderFormValues extends OpenIDProvider {}
+
+export const defaultInitialValues: OpenIDProviderFormValues = {
+  id: "",
+  provider: "",
+  client_id: "",
+  client_secret: "",
+};
+
+export const transformOrganizationToFormValues = (
+  openIDProvider: OpenIDProvider
+): OpenIDProviderFormValues => ({ ...openIDProvider });
+
+export const transformFormValuesToOpenIDProvider = (
```

Comment:
If these have all the same properties, I think we can remove the need to have different types and transforming between them. You can also cast types, let me know if you have questions or you want to have a short call to review it together.